### ISSUE TITLE
False analysis result of U2U1015

### ISSUE THREAD

**Issue by wmjordan** [u2uconsult/codeanalyzers#4]
The following code triggers a false warning of U2U1015.
````C#
for (int i = 0; i < a.Length; i++) {
	a[i] = a[i].TrimStart(' ');
}
````

It is totally valid to read from the array and put a manipulated value back to that place and should not trigger that warning.

**Comment by KrisVandermotten** [comment]
Of course the code is totally valid. But it can be optimized. That is why we generate the warning "Array a is indexed multiple times within a loop body, **consider** using a ref local variable"

Your code can be optimized, for example:

    for (int i = 0; i < a.Length; i++)
    {
        ref string p = ref a[i];

        p = p.TrimStart(' ');
    }

That being said, I doubt that this optimization will even be measurable in your example, given that you are calling `TrimStart` for each item.

For more information, see [U2U1015](/u2uconsult/codeanalyzers/blob/master/docs/U2U1015.md).

**Comment by wmjordan** [comment]
Never expected that.
Thanks for the clarification.
